docs/life_of_a_godwoken_transaction.md

```diff
@@ -0,0 +1,338 @@
+
+```json
+0xdab9fff9ede40ee4c4f77fea4af996e1b12cc0c4520e36eec11ecc0097a96baf33000000e1a06940a4a5f105ed0ef1c6d8bf5878c793aa62904bd1cb75431186587a57488b420000000000000000000000000000000000000000000000000000000000000000000000000000a23e00000000000000
+```
+
+This is actually a `GlobalState` data structure serialized in [molecule](https://github.com/nervosnetwork/molecule)'s format:
+
+```
+struct GlobalState {
+    account: AccountMerkleState,
+    block: BlockMerkleState,
+    reverted_block_root: Byte32,
+    last_finalized_block_number: Uint64,
+    // 0: running, 1: halting
+    status: byte,
+}
+```
+
+Please refer to [this file](https://github.com/nervosnetwork/godwoken/blob/7a9d0e594e2fe80ea9bd4b7c013dc705f4e9f05d/crates/types/schemas/godwoken.mol#L18) for more definitions.
+
+The actual layer 2 Godwoken block, is also contained in the layer 1 CKB transaction. The first witness of the above mentioned transaction, contains the following data:
+
+```
+0x2c02000010000000100000001000000018020000180200002000000018010000590100005d010000610100006501000014020000f8000000240000002c0000003000000050000000580000007c000000a0000000d80000008a42000000000000000000000000000000000000000000000000000000000000000000000000000000000000e1870e6000000000dab9fff9ede40ee4c4f77fea4af996e1b12cc0c4520e36eec11ecc0097a96baf33000000dab9fff9ede40ee4c4f77fea4af996e1b12cc0c4520e36eec11ecc0097a96baf33000000380000001000000030000000340000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000040000000000000004000000ab0000004c50015a173879f96ed3d518764255849a34282b3b00326848c7e9e5c4b70dee4e34b750030db70edcd84c516ebe6b9e2b8c37f93001a740e20305f061147b91f5e8b078e85007e8deaaae54dc3d560e4da736fd8ac06d6b1e60d98fa6ee475524a63ca1b3b9aa5009d719c2db9dd31513c5d0a3cd22012400b55bb2fe91bf347c378ba4ac7379fc52500e1ca9d7c1150f156443c30a3581721d99915d6318c57b6af459aae3781e92147800000000
+```
+
+This is actually a `WitnessArgs` data structure described [here](https://github.com/nervosnetwork/godwoken/blob/7a9d0e594e2fe80ea9bd4b7c013dc705f4e9f05d/crates/types/schemas/blockchain.mol#L105), and serialized in [molecule](https://github.com/nervosnetwork/molecule)'s format. In the `output_type` field of WitnessArgs, the actual `L2Block` data structure is stored, also serialized in molecule's format:
+
+```
+table L2Block {
+    raw: RawL2Block,
+    signature: Signature,
```

Comment:
This field will be removed soon.

Reply:
This post is now written against v0.1.0 version, we can update those data structure when the new version is released.